### PR TITLE
[SPARK-45795][SQL] DS V2 supports push down Mode

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR will translate the aggregate function `MODE` for pushdown.

The constructor of aggregate function `MODE` has a `deterministic` parameter. When multiple values have the same greatest frequency then either any of values is returned if `deterministic` is false or is not defined, or the lowest value is returned if `deterministic` is true.
If `deterministic` is true, the semantics of `deterministic` is the same as the syntax supported by some databases (e.g. H2, Postgres) show below.
The syntax is:
`MODE() WITHIN GROUP (ORDER BY col)`.

Note: `MODE() WITHIN GROUP (ORDER BY col)` doesn't support `DISTINCT` keyword.

### Why are the changes needed?
DS V2 supports push down `Mode`

### Does this PR introduce _any_ user-facing change?
'No'.
New feature.

### How was this patch tested?
New test cases.

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @huaxingao @cloud-fan

**Review by HyukjinKwon — APPROVED** [review]

**Comment by beliefer** [comment]
@huaxingao @cloud-fan

**Comment by beliefer** [comment]
Merged to master.
@HyukjinKwon Thank you!

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/aggregate/GeneralAggregateFunc.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
  *  <li><pre>REGR_R2(input1, input2)</pre> Since 3.4.0</li>
  *  <li><pre>REGR_SLOPE(input1, input2)</pre> Since 3.4.0</li>
  *  <li><pre>REGR_SXY(input1, input2)</pre> Since 3.4.0</li>
+ *  <li><pre>MODE(input1[, inverse])</pre> Since 4.0.0</li>
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/aggregate/GeneralAggregateFunc.java:44]: This is very hacky. It's kind of we re-define the SQL semantic of the second parameter for MODE function only. Do we have existing examples?

Comment (beliefer) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/aggregate/GeneralAggregateFunc.java:44]: Got it. Let's improve it.

**Pull request closed without merging**